### PR TITLE
handle unregistered migrations in up-to-date check

### PR DESCRIPTION
**Pull request by dbu**
|      Q       |   A
|------------- | -----------
| Type         | bug & improvement
| BC Break     | no
| Fixed issues | -

#### Summary

We need to handle the case where there are extra migrations in doctrine. without this change, the command outputs something like `"Out-of-date! 18446744073709551613 migration is available to execute."` we convert the signed integer value to unsigned.

with this fix, we explicitly handle the case. the default is to report the issue but not fail, same as the migrate command that only outputs a warning but does not refuse to run. i also add a flag in case people want a strict check.

### CODE REVIEW

File: lib/Doctrine/Migrations/Tools/Console/Command/UpToDateCommand.php [hunk 1]
```diff
@@ -43,12 +45,26 @@ public function execute(InputInterface $input, OutputInterface $output) : ?int
...
-            $availableMigrations,
-            $availableMigrations > 1 ? 's are' : ' is'
+            '<error>You have %1$u previously executed migration%3$s in the database that %2$s registered migration%3$s.</error>',
+            $extraMigrations,
+            $extraMigrations > 1 ? 'are not' : 'is not a',
+            $extraMigrations > 1 ? 's' : ''
         ));
 
-        return 1;
+        return true === $input->getOption('fail-on-unregistered') ? 2 : 0;
```
Comment (dbu) [inline lib/Doctrine/Migrations/Tools/Console/Command/UpToDateCommand.php]: i originally had `$input->getOption('fail-on-unregistered') ? 2 : 0` but then the code sniffer complained that the return value of getOption is mixed. i tried to fix it like this, but now the code style check complains about yoda style comparison. how should i do the comparison?

Comment (SenseException) [reply lib/Doctrine/Migrations/Tools/Console/Command/UpToDateCommand.php]: `$input->getOption('fail-on-unregistered') === true`

**Comment by dbu** [comment]
looks like scrutinizer failed talking to github or something.

is that ok as is, should i adjust anything with the behaviour?

**Comment by jwage** [comment]
Usually we just restart the build and try it again.

**Comment by dbu** [comment]
i don't think i have permission to do that.

content-wise, do you need anything more to have this mergeable? changelog entry maybe? and do you agree with the actual feature addition?

**Pull request merged**